src/mpid/common/thread/mpidu_thread_fallback.h

```diff
@@ -394,117 +306,61 @@ M*/
             /* silence warnings: -Wmaybe-uninitialized */               \
             st.owner = 0;                                               \
             st.count = 0;                                               \
-        } \
+        }                                                               \
     } while (0)
 
-#if MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__GLOBAL
-
-#define MPIDUI_THREAD_CS_EXIT_ST_GLOBAL    MPIDUI_THREAD_CS_EXIT_ST
-#define MPIDUI_THREAD_CS_EXIT_ST_POBJ(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_EXIT_ST_VCI_GLOBAL MPIDUI_THREAD_CS_EXIT_GLOBAL
-#define MPIDUI_THREAD_CS_EXIT_ST_VCI(mutex) do {} while (0)
-
-#elif MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__POBJ
-
-#define MPIDUI_THREAD_CS_EXIT_ST_POBJ(mutex)  MPIDUI_THREAD_CS_EXIT_ST(mutex)
-#define MPIDUI_THREAD_CS_EXIT_ST_GLOBAL(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_EXIT_ST_VCI_GLOBAL(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_EXIT_ST_VCI(mutex) do {} while (0)
-
-#elif MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__VCI
-
-#define MPIDUI_THREAD_CS_EXIT_ST_GLOBAL MPIDUI_THREAD_CS_EXIT_ST
-#define MPIDUI_THREAD_CS_EXIT_ST_POBJ(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_EXIT_ST_VCI_GLOBAL(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_EXIT_ST_VCI(mutex) MPIDUI_THREAD_CS_EXIT_ST(mutex)
-
-#endif /* MPICH_THREAD_GRANULARITY */
-
-#else /* !defined(MPICH_IS_THREADED) */
-
-#define MPIDUI_THREAD_CS_ENTER_ST_GLOBAL(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_ENTER_ST_POBJ(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_ENTER_ST_VCI_GLOBAL(mutex) do {} while (0)
-#define MPIDUI_THREAD_CS_ENTER_ST_VCI(mutex) do {} while (0)
-
 #endif /* MPICH_IS_THREADED */
 
-/*M MPIDU_THREAD_CS_YIELD - Temporarily release a critical section and yield
-    to other threads
-
-  Input Parameters:
-+ _name - cname of the critical section
-- _context - A context (typically an object) of the critical section
-
-  M*/
-#define MPIDU_THREAD_CS_YIELD(name, mutex) MPIDUI_THREAD_CS_YIELD_##name(mutex)
+/* MPICH_THREAD_GRANULARITY (set via `--enable-thread-cs=...`) activates one set of locks */
 
 #if defined(MPICH_IS_THREADED)
```

Comment:
This `#if defined(MPICH_IS_THREADED)` can be removed since it is just following an `#endif` that checks for the same.

Reply:
done